tasks/ms_defender_tvm/lib/ms_defender_tvm_helper.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: true
+
+module Kenna
+  module Toolkit
+    module MSDefenderTvmHelper
+      @client_id = nil
+      @tenant_id = nil
+      @client_secret = nil
+      @tvm_query_api = nil
+      @tvm_oath_url = nil
+      @token = nil
+      @uploaded_files = nil
+      @file_cleanup = nil
+
+      def connector_upload(output_dir, filename, kenna_connector_id, kenna_api_host, kenna_api_key, max_retries = 3)
+        ### Write KDI format
+        kdi_output = { skip_autoclose: false, assets: @paged_assets, vuln_defs: @vuln_defs }
+        write_file output_dir, filename, JSON.pretty_generate(kdi_output)
+        print_good "Output is available at: #{filename}"
+
+        ### Finish by uploading if we're all configured
+        if kenna_connector_id && kenna_api_host && kenna_api_key
+          print_good "Attempting to upload to Kenna API at #{kenna_api_host}"
+          response_json = upload_file_to_kenna_connector kenna_connector_id, kenna_api_host, kenna_api_key, "#{output_dir}/#{filename}", false, max_retries
+          filenum = response_json.fetch("data_file")
+          @uploaded_files = [] if @uploaded_files.nil?
+          @uploaded_files << filenum
+          File.delete("#{output_dir}/#{filename}") if @file_cleanup
+        end
+        response_json
+      end
+
+      def connector_kickoff(kenna_connector_id, kenna_api_host, kenna_api_key, max_retries = 3)
+        ### Finish by uploading if we're all configured
+        return unless kenna_connector_id && kenna_api_host && kenna_api_key
+
+        print_good "Attempting to run to Kenna Connector at #{kenna_api_host}"
+        run_files_on_kenna_connector kenna_connector_id, kenna_api_host, kenna_api_key, @uploaded_files, max_retries
+      end
+
+      def tvm_get_machines(page_param = nil)
+        print_debug "Getting machines"
```

Comment:
Do you want this and other print_debug() statements left in the code?

Reply:
yeah, they will only print out if debug=true so in most cases users will never see them.